### ISSUE TITLE
Query Loop block: Consider click to edit treatment

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#46955]
## What problem does this address?

This is feedback from the [nineteenth call for testing for the FSE Outreach Program](https://make.wordpress.org/test/2022/12/14/fse-program-testing-call-19-running-through-refinements/#comment-2861): 

>There is no easy way to select the Query Loop without going to the List View, as any attempt to select an element on that block selects the Column ‘level’ or deeper.

Here's a quick video with GB 14.9 to demonstrate how you both can select the query loop block but also easily miss it:

https://user-images.githubusercontent.com/26996883/211110022-9f3566cf-4a65-4a2a-800e-c518359367a4.mov

## What is your proposed solution?

I am torn here as I both like the experience as it is and could see where, upon emphasizing patterns more, we might want to consider a "click to edit" treatment so folks can replace/swap patterns with more ease. What about if a site has patterns turned off? Do we want to push folks into the complexity of the query loop block outright? I also have repeatedly seen feedback about it being annoying to double click the navigation block to access it due to this same "click to edit" treatment. In that case, it makes sense to me since we do need an easy way for folks to select a header. 

Curious what @WordPress/gutenberg-design  thinks :)

**Comment by jameskoster** [comment]
> There is no easy way to select the Query Loop without going to the List View, as any attempt to select an element on that block selects the Column ‘level’ or deeper.

Worth noting this is not limited to the Query Loop, and is pretty much the case for all containers. That is to say, would it be strange if Query Loop behaved differently to other container blocks in that respect, or should we consider all containers here?

I wonder if there's something we could do with the locking api to make this behavior a user preference... perhaps an option that disables on-canvas selection of a containers children? Or even simpler: blocks in a container that are locked cannot be selected on-canvas?

**Comment by annezazu** [comment]
Agreed. This feels a bit like a slippery slope but I'm not sure how it also ties into integrating with patterns as I imagine container blocks are where the pattern switching/swapping will happen. A user preference might be a neat approach to this :)

**Comment by jameskoster** [comment]
Spitballing some more... If there was a "Disable editing" option here:

<img width="510" alt="Screenshot 2023-01-10 at 11 09 53" src="https://user-images.githubusercontent.com/846565/211536127-4e4b1b1b-1e9d-4513-bef9-20d6d8a7ca4c.png">

And all the blocks inside a container were fully locked, then it seems plausible that those blocks would not be selectable on the canvas, and a 'click-to-edit' UI might be organically surfaced when the parent container is selected.

In terms of pattern switching, I think that falls into `"templateLock":"contentOnly"` territory, and is being worked on here: https://github.com/WordPress/gutenberg/pull/45618. We still need to figure out how that integrates with the locking api too :)

**Comment by ntsekouras** [comment]
Thanks for raising this! The discussion here moved toward a generic solution for containers rather than a Query Loop specific treatment, and that direction has since shipped as content-only section editing and the Write mode. I'm closing this one, but happy to reopen if design wants to revisit a click to edit treatment for the block.
